OrbitGl/TimeGraph.cpp

```diff
@@ -956,6 +979,32 @@ void TimeGraph::SortTracks() {
       sorted_tracks_ = std::move(filtered_tracks);
     }
 
+    // Separate "capture_pid" tracks from tracks that originate from other processes.
+    int32_t capture_pid = GOrbitApp->GetCaptureData().process_id();
+    std::vector<std::shared_ptr<Track>> capture_pid_tracks;
+    std::vector<std::shared_ptr<Track>> external_pid_tracks;
+    for (auto& track : sorted_tracks_) {
+      int32_t pid = track->GetProcessId();
+      if (pid != -1 && pid != capture_pid) {
+        external_pid_tracks.emplace_back(track);
+      } else {
+        capture_pid_tracks.emplace_back(track);
+      }
+    }
+
+    // Clear before repopulating.
+    sorted_tracks_.clear();
+
+    // Scheduler track.
+    if (!scheduler_track_->IsEmpty()) {
+      sorted_tracks_.emplace_back(scheduler_track_);
+    }
+
+    // For now, "external_pid_tracks" should only contain
+    // introspection tracks. Display them on top.
+    Append(sorted_tracks_, external_pid_tracks);
+    Append(sorted_tracks_, capture_pid_tracks);
```

Comment:
Not in favor of these utility functions, `std::insert` should be used.

Reply:
I find Append is much more verbose and concise than the insert syntax. It would take two lines to write using insert!!

Reply:
```
std::insert(sorted_tracks_.begin(), external_pid_tracks.begin(), external_pid_tracks.end());
std::insert(sorted_tracks_.begin(), capture_pid_tracks.begin(), capture_pid_tracks.end());
```

Sure, you have to write the source twice, but it's a well-known function from the standard, used in many places in the codebase. Having a wrapper around a standard function for such minor overhead is in my opinion not a good practice.